### PR TITLE
Add support for varbinary filters in Pinot

### PR DESCRIPTION
**Pull request by elonazoulay**
Also includes fixes to real and double decoding for broker requests.

### CODE REVIEW

**Comment by hashhar** [comment]
Related test failure:

```
Error:  io.trino.plugin.pinot.TestPinotIntegrationSmokeTest.testNullBehavior  Time elapsed: 9.491 s  <<< FAILURE!
java.lang.AssertionError: 
Plan does not match, expected [

- anyTree
    - node(ExchangeNode)
        - node(ProjectNode)
            - node(TableScanNode)

] but found [

Output[_col0]
│   Layout: [element_at:varchar]
│   _col0 := element_at
└─ RemoteExchange[GATHER]
   │   Layout: [element_at:varchar]
   └─ ScanFilterProject[table = pinot:PinotTableHandle{schemaName=default, tableName=alltypes, constraint={PinotColumnHandle{columnName=bytes_col, dataType=varbinary, returnNullOnEmptyGroup=true}=[ SortedRangeSet[type=varbinary, ranges=1, {[Slice{address=0, length=0}]}] ]}, limit=OptionalLong.empty, query=Optional.empty}, filterPredicate = (element_at("bool_array_col", BIGINT '1') = VARCHAR 'null')]
          Layout: [element_at:varchar]
          element_at := element_at("string_array_col", BIGINT '1')
          bool_array_col := PinotColumnHandle{columnName=bool_array_col, dataType=array(varchar), returnNullOnEmptyGroup=true}
          string_array_col := PinotColumnHandle{columnName=string_array_col, dataType=array(varchar), returnNullOnEmptyGroup=true}

]
	at io.trino.sql.planner.assertions.PlanAssert.assertPlan(PlanAssert.java:56)
	at io.trino.sql.planner.assertions.PlanAssert.assertPlan(PlanAssert.java:47)
	at io.trino.sql.planner.assertions.PlanAssert.assertPlan(PlanAssert.java:41)
	at io.trino.sql.query.QueryAssertions$QueryAssert.lambda$isNotFullyPushedDown$9(QueryAssertions.java:466)
	at io.trino.transaction.TransactionBuilder.lambda$execute$1(TransactionBuilder.java:122)
	at io.trino.transaction.TransactionBuilder.execute(TransactionBuilder.java:150)
	at io.trino.transaction.TransactionBuilder.execute(TransactionBuilder.java:121)
	at io.trino.sql.query.QueryAssertions$QueryAssert.isNotFullyPushedDown(QueryAssertions.java:464)
	at io.trino.sql.query.QueryAssertions$QueryAssert.isNotFullyPushedDown(QueryAssertions.java:450)
	at io.trino.plugin.pinot.TestPinotIntegrationSmokeTest.testNullBehavior(TestPinotIntegrationSmokeTest.java:753)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by findepi** [comment]
Can you consider adding "in Pinot" in commit messages?
they are short enough that this can be added without going over the limit

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java [hunk 1]
```diff
@@ -25,6 +25,8 @@
 import io.trino.spi.type.RealType;
```
Comment (hashhar) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/PinotQueryBuilder.java:25]: A truer commit message would be `Add support for pushdown of varbinary filters to Pinot`.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/RealDecoder.java [hunk 2]
```diff
@@ -29,6 +29,10 @@ public void decode(Supplier<Object> getter, BlockBuilder output)
         if (value == null) {
             output.appendNull();
         }
+        else if (value instanceof String) {
+            // Pinot returns NEGATIVE_INFINITY, POSITIVE_INFINITY as a String
+            output.writeInt((floatToIntBits(Float.valueOf(value.toString()))));
```
Comment (hashhar) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/RealDecoder.java:34]: redundant parenthesis?

Comment (hashhar) [reply plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/RealDecoder.java:34]: Also if `value` is a String is the `value.toString()` really needed? A cast should do.

File: plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/DoubleDecoder.java [hunk 3]
```diff
@@ -29,6 +29,10 @@ public void decode(Supplier<Object> getter, BlockBuilder output)
         if (value == null) {
             output.appendNull();
         }
+        else if (value instanceof String) {
+            // Pinot returns NEGATIVE_INFINITY, POSITIVE_INFINITY as a String
+            DOUBLE.writeDouble(output, Double.valueOf(value.toString()));
```
Comment (hashhar) [inline plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/decoders/DoubleDecoder.java:34]: Also if `value` is a String is the `value.toString()` really needed? A cast should do.

**Review by hashhar — APPROVED** [review]
LGTM % comments.

**Pull request merged**